academic-services/knowledge-exploration-service/reference-get-interpret.md

```diff
@@ -0,0 +1,459 @@
+---
+title: GET Interpret
+description: Generates semantic interpretations of a natural language query
+ms.topic: reference
+ms.date: 2020-02-07
+---
+
+# Interpret REST API
+
+The **Interpret** method generates semantic interpretations of a natural language query.
+
+It can be used as either a way of interpreting a specific natural language query, or to generate natural language query suggestions using the original query as a stem.
+
+For full details on the language grammar see the [natural language queries](concepts-queries.md) page.
+
+``` HTTP
+GET http://{serviceName}.{serviceRegion}.cloudapp.azure.com/interpret?query={query}
+```  
+
+With optional parameters:
+
+``` HTTP
+GET http://{serviceName}.{serviceRegion}.cloudapp.azure.com/interpret?query={query}&count={count}&offset={offset}&timeout={timeout}&complete={complete}&entityCount={entityCount}&attributes={attributes}
+```  
+
+## URI Parameters
+
+Name | Required | Type | Description
+--- | --- | --- | ---
+`query` | Required | string | The natural language query to generate interpretations for. See the [natural language queries](concepts-queries.md) page for more details.
+`count` | Optional | integer | The maximum number of interpretations to generate. <br/><br/>Defaults to 5.
+`complete` | Optional | integer | A value of 1 indicates that interpretations should be generated that infer entity attribute values beyond what was provided in the query. <br/><br/>For example, the for query "microsoft machine le", if `complete` is set to 1 an interpretation could be generated that completes "le" to "learning", i.e. "microsoft machine learning". <br/><br/>A value of 0 indicates that no inference is made and only the exact query text provided is matched. <br/><br/>Defaults to 1.
+`normalize` | Optional | integer | Value of 1 indicates that the query should have normalization rules applied to it before being interpreted. See the [entity schema overview](reference-makes-api-entity-schema.md#normalization-rules) for documentation on the normalization rules that are applied to indexed string fields. <br/><br/>Defaults to 1.
+`offset` | Optional | integer | The number of interpretations to skip in the result set. <br/><br/>Defaults to 0.
+`timeout` | Optional | integer | The maximum amount of time in milliseconds to use when generating interpretations. <br/><br/>If the timeout is hit, all interpretations that have been generated will be returned and a "timedOut" flag will be set in the response indicating that the timeout was hit before the total number of requested interpretations was met. <br/><br/>Defaults to 2000.
+`entityCount` | Optional | integer | The number of entities that should be returned for each interpretation that match the interpreted query expression. <br/><br/>Defaults to 0.
```

Comment:
>Defaults to 0 [](start = 163, length = 13)

Default to 0. Does it means no entity will be returned?

Reply:
Correct

---
In reply to: [373616233](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/97#discussion_r373616233) [](ancestors = 373616233)